tests/linearizability/model/model.go

```diff
@@ -288,7 +298,9 @@ func initState(request EtcdRequest, response EtcdResponse) EtcdState {
 func applyFailedRequest(states PossibleStates, request EtcdRequest) PossibleStates {
 	for _, s := range states {
 		newState, _ := applyRequestToSingleState(s, request)
-		states = append(states, newState)
+		if !reflect.DeepEqual(newState, s) {
+			states = append(states, newState)
+		}
```

Comment:
Could you clarify this change? Shouldn't we always cache the state for failed request?

Reply:
I'm adding an exception because of the new Defragment request. As defragment should not impact etcd results, the etcd state does not change. If etcd state does not change, there is no need to "cache" the state on failed request.